sys/src/9/port/sysfile.c

```diff
@@ -775,12 +774,12 @@ read(int fd, void *p, int32_t n, int64_t off)
 		}
 		nnn = mountfix(c, p, nn, n);
 
-		sequential = 1;
+		dopread = 1;
 	}
 	else
 		nnn = nn = c->dev->read(c, p, n, off);
 
-	if(sequential){
+	if(dopread){
```

Comment:
This is reversed; but I think you can just always update the offsets in every case.

    lock(&c->r.l);
    if(nn > 0)
        c->devoffset += nn;
    if (nnn > 0)
        c->offset += nnn;
    unlock(&c->r.l);